docs/guides/observability-metrics.md

```diff
@@ -0,0 +1,350 @@
+
+Here is an overview of the types of metrics that are exposed by TriggerMesh components. The list is deliberately broad
+and generic, as some categories of metrics may only be exposed by certain types of components.
+
+- Processing of events
+    - Successes and errors
+    - Latency distribution
+- Delivery of events
+    - Successes and errors
+    - Latency distribution
+- Software runtime
+    - Heap memory usage
+    - Garbage collection
+
+Later in this document, we will explore how metrics in these different categories can be collected, analyzed and
+visualized.
+
+### Data Model
+
+Metrics are exposed by TriggerMesh in a line-oriented [text-based format][prom-metrics-fmt] popularized by the
+[Prometheus][prom-overview] open source monitoring toolkit.
+
+A Prometheus metric is typically represented as a single line of UTF-8 characters, optionally prepended with a `HELP`
+and a `TYPE` comment lines, starting with the **name** of the metric and ending with its **value**:
+
+```openmetrics
+# HELP event_processing_success_count The number of events successfully processed
+# TYPE event_processing_success_count counter
+event_processing_success_count{event_source="my.cloud.storage", event_type="object.created"} 129389
+```
+
+The `HELP` comment provides a description of _what_ the metric represents, whereas the `TYPE` comment carries
+information about the _type_ of the metric (one of the four [core metric types][prom-metrics-typ] offered by
+Prometheus).
+
+As illustrated in the example above, the metric name may be directly followed by a list of comma-separated key-value
+pairs between curly brackets called **labels**. Labels allow differentiating the characteristics of what is being
+measured. Each unique combination of labels identifies a particular **dimension** of a metric. Labels, and metric
+dimensions by extension, vary from metric to metric.
+
+### Access to Metrics
+
+Every TriggerMesh component exposes a set of telemetry metrics on the local HTTP endpoint `:9092/metrics`. Available
+measurements can be retrieved via a simple `GET` request. The returned values are an **instant view** of each
+measurement at the time of the request.
+
+This model is particularly suitable for a pull-based retrieval of metrics ("scrape") on a fixed interval by monitoring
+software, and for storage in a time-series database. Prometheus itself is a popular choice for this job, as it includes
+both a server for scraping telemetry metrics, and a time-series database optimized for the Prometheus data model. 
+
+![Metrics scraping](../assets/guides/observability/metrics-scrape.excalidraw.png)
+
+Most of the available metrics fall into the categories described previously in this document, whenever applicable based
+on the type of component. Some components may expose additional, application-specific metrics. The data model presented
+in the previous section makes these metrics easily discoverable by consumers.
+
+## Collection and Analysis 
```

Comment:
We could add "with Prometheus and Grafana" to the title to show this section is a practical example?

Reply:
Addressed in [ca6da7f5..59acc041](https://github.com/triggermesh/docs/compare/ca6da7f5eb00e390c91b11b9256d1493332c074b..59acc04112860984ea0596b416e97db5f3680955)